importer/helper.go

```diff
@@ -252,29 +252,19 @@ func InitialImport(db idb.IndexerDb, genesisJSONPath string, client *algod.Clien
 	//var genesisString string
 	var genesisReader io.Reader
 
-	// Attempt to read file.
+	// Attempt to read file if specified.
 	if genesisJSONPath != "" {
 		l.Infof("loading genesis file %s", genesisJSONPath)
 		genesisReader, err = os.Open(genesisJSONPath)
-		if err != nil {
-			genesisReader = nil
-		}
+		maybeFail(err, l, "unable to read genesis file %s", genesisJSONPath)
 	}
 
 	// If we still don't have it but we have a client, ask algod.
 	if client != nil && genesisReader == nil {
 		l.Infof("fetching genesis from algod")
 		genesisString, err := client.GetGenesis().Do(context.Background())
-		if err != nil {
-			genesisReader = nil
-		} else {
-			genesisReader = strings.NewReader(genesisString)
-		}
-	}
-
-	if genesisReader == nil {
-		l.Error("Failed to get genesis data.")
-		os.Exit(1)
+		maybeFail(err, l, "unable to fetch genesis from algod")
+		genesisReader = strings.NewReader(genesisString)
 	}
 
 	err = loadGenesis(db, genesisReader)
```

Comment:
If filepath and algod are both not provided, this will return a strange error. I would do this:
```
if (path != nil) || (client != nil) {
  if path != nil {
  ...
  } else {
  ...
  }
  loadGenesis()
} else {
  return error
}
```